src/effects/native/phasereffect.cpp

```diff
@@ -115,13 +129,18 @@ void PhaserEffect::processChannel(const ChannelHandle& handle,
                                   const unsigned int sampleRate,
                                   const EffectProcessor::EnableState enableState,
                                   const GroupFeatureState& groupFeatures) {
-
     Q_UNUSED(handle);
     Q_UNUSED(enableState);
-    Q_UNUSED(groupFeatures);
-    Q_UNUSED(sampleRate);
 
-    CSAMPLE frequency = m_pLFOFrequencyParameter->value();
+    CSAMPLE period = std::max(m_pLFOPeriodParameter->value(), 0.05);
+    if (m_pSyncParameter->toBool() && groupFeatures.has_beat_length) {
+        // Period is a number of beats
+        period = period * groupFeatures.beat_length;
+    } else {
+        // Period is a number of seconds
+        // TODO: remove assumption of stereo signal
+        period = period * sampleRate * 2;
```

Comment:
Picking any BPM would be totally arbitrary and inappropriate for a lot of music. Using the master clock would make sense, but the effects system does not provide that information currently. Only information about the channel being processed is available to the effect.